app/src/sharedTest/java/org/oppia/app/player/audio/CellularDataDialogFragmentTest.kt

```diff
@@ -0,0 +1,72 @@
+package org.oppia.app.player.audio
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.home.HomeActivity
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
+/** Tests for [CellularDataDialogFragment]. */
+@RunWith(AndroidJUnit4::class)
+class CellularDataDialogFragmentTest {
```

Comment:
Suggest also adding a TODO against #116 to properly test the flows where the user enables/disables on cellular with/without saving the setting (& for verifying that saving the setting works correctly).